### ISSUE TITLE
Cached resolution: integration tests do not see libraries from dependencies with caching

### ISSUE THREAD

**Issue by fommil** [sbt/sbt#1868]
This only appears when using

``` scala
updateOptions := updateOptions.value.withCachedResolution(true),
```

If I have a `Project` with an `IntegrationTest` configuration enabled, the libraries that are pulled in by dependsOn projects are not on the compile path of the integration tests.

Workaround is to disable the 0.13.7 caching or to list all the dependencies again.

I have to use 0.13.7-RC3 because the introduction of #1776 stops me from upgrading past that point.

**Comment by fommil** [comment]
wow, it's even worse than I expected. Even the `libraryDependencies` defined in the common settings don't work.

**Comment by fommil** [comment]
still present in 0.13.8-M5

**Comment by eed3si9n** [comment]
This is still an issue with 0.13.8-M6 as well.

**Comment by fommil** [comment]
cool, thanks for checking. I was just about to let you know.

**Comment by fommil** [comment]
confirmed fixed in 0.13.8

**Comment by devshorts** [comment]
I'm intermittently seeing this in sbt 0.13.13. Were there fixes to this in .15/.16 that are worth trying?  I'm seeing that _sometimes_ a library isn't visible to the compilation stage when its code is referenced in an integration test in the `it` folder.  Adding the dependency explicity to `it` does make it work.

I checked the visible dependencies in sbt for the config via 
```
show it:libraryDependencies
```
and compared to 
```
show libraryDependencies
```
And the main ones (minus explicit `it` designated libs) were the same, so I'm not sure why the compilation would fail. Removing the cache resolution sbt toggle resolves the issue
